src/retrace/read_config.py

```diff
@@ -0,0 +1,115 @@
+#Singleton class for saving path to the configuration file
+
+#Heavily inspired by
+#   http://python-3-patterns-idioms-test.readthedocs.io/en/latest/Singleton.html
+#All modules that need reading from configuration file must create instance of
+#   ConfigClass. When different path than default is needed, before first
+#   reading from any CONFIG item, a read_config(/path/to/the/config/file)
+#   must be called. When this function was not called, is automatically called
+#   before first reading from CONFIG with default path.
+#Note:all modules share one instance, therefore only one read_config is needed.
+
+import ConfigParser
+
+class ConfigClass(object):
+    conf_file_read = False
+    class __config_class:
+        CONFIG = {
+          "TaskIdLength": 9,
+          "TaskPassLength": 32,
+          "MaxParallelTasks": 10,
+          "MaxPackedSize": 30,
+          "MaxUnpackedSize": 600,
+          "MinStorageLeft": 10240,
+          "DeleteTaskAfter": 120,
+          "DeleteFailedTaskAfter": 24,
+          "ArchiveTaskAfter": 0,
+          "KeepRawhideLatest": 3,
+          "KojiRoot": "/mnt/koji",
+          "DropDir": "/srv/retrace/archive",
+          "LogDir": "/var/log/retrace-server",
+          "RepoDir": "/var/cache/retrace-server",
+          "SaveDir": "/var/spool/retrace-server",
+          "WorkDir": "/tmp/retrace-server",
+          "UseWorkDir": False,
+          "RequireHTTPS": True,
+          "AllowAPIDelete": False,
+          "AllowExternalDir": False,
+          "AllowInteractive": False,
+          "AllowTaskManager": False,
+          "AllowVMCoreTask" : False,
+          "AllowUsrCoreTask" : False,
+          "TaskManagerAuthDelete": False,
+          "TaskManagerDeleteUsers": [],
+          "UseFTPTasks": False,
+          "FTPSSL": False,
+          "FTPHost": "",
+          "FTPUser": "",
+          "FTPPass": "",
+          "FTPDir": "/",
+          "FTPBufferSize": 16,
+          "WgetKernelDebuginfos": False,
+          "KernelDebuginfoURL": "http://kojipkgs.fedoraproject.org/packages/kernel/$VERSION/$RELEASE/$ARCH/",
+          "VmcoreDumpLevel": 0,
+          "VmcoreRunKmem": 0,
+          "RequireGPGCheck": True,
+          "UseCreaterepoUpdate": False,
+          "DBFile": "stats.db",
+          "KernelChrootRepo": "http://dl.fedoraproject.org/pub/fedora/linux/releases/16/Everything/$ARCH/os/",
+          "UseFafPackages": False,
+          "FafLinkDir": "/var/spool/faf/retrace-tmp",
+          "AuthGroup": "retrace",
+          "EmailNotify": False,
+          "EmailNotifyFrom": "retrace@localhost",
+          "CaseNumberURL": "",
+          "Crashi386": "",
+        }
+
+        def __getitem__(self, key):
```

Comment:
What is the purpose of this method? I believe you wanted to do something like this:
```
def __getitem__(self, key):
     if not self.conf_file_read:
            self.read_config()
     return CONF[key]
```